yew/src/virtual_dom/vtag.rs

```diff
@@ -110,6 +119,80 @@ impl Clone for VTag {
     }
 }
 
+/// HTML output for a VTag is not necessarily deterministic due to the
+/// serialization of props which do not have a particular ordering.
+#[cfg(feature = "sans_mount_render")]
+impl TryFrom<VTag> for Html {
+    type Error = HtmlRenderError;
+
+    fn try_from(value: VTag) -> Result<Html, HtmlRenderError> {
+        let mut result: String = "".to_string();
+        let tag_name = htmlescape::encode_minimal(&value.tag).to_lowercase();
+
+        if !is_valid_sgml_tag(tag_name.as_ref()) {
+            return Err(HtmlRenderError::InvalidTagName(tag_name));
+        }
+
+        result.push_str(format!("<{}", tag_name).as_ref());
+
+        for (key_unclean, value) in &value.attributes {
+            let key = key_unclean.to_lowercase();
+            // checked, value (special if textarea), disabled, href?, selected,
+            // kind -> type if input, disallow ref, disallow LISTENER_SET, class
+
+            if !is_valid_html_attribute_name(key.as_str()) {
+                return Err(HtmlRenderError::InvalidAttributeName(key));
+            }
+
+            // textareas' innerHTML properties are specified via the `value` prop which doesn't
+            // exist in HTML, so we defer this prop's serialization until later in the process.
+            if tag_name == "textarea" && key == "value" {
+                continue;
+            }
+
+            result.push_str(
+                format!(
+                    " {}=\"{}\"",
+                    htmlescape::encode_minimal(&key),
+                    htmlescape::encode_attribute(&value)
+                )
+                .as_ref(),
+            );
+        }
+
+        if value.checked {
+            result.push_str(" checked=\"true\"")
```

Comment:
For boolean attributes the spec has this to say:
> If the attribute is present, its value must either be the empty string or a value that is an ASCII case-insensitive match for the attribute's canonical name, with no leading or trailing whitespace.

[source](https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#boolean-attribute)

I prefer the empty string version:
```suggestion
            write!(result, " checked=\"\"");
```

Reply:
Thanks for looking up the spec on this.